sdk/apis/apis/v1alpha1/types_apiexportendpointslice.go

```diff
@@ -106,7 +121,10 @@ const (
 	// PartitionValid is a condition for APIExportEndpointSlice that reflects the validity of the referenced Partition.
 	PartitionValid conditionsv1alpha1.ConditionType = "PartitionValid"
 
-	APIExportEndpointSliceURLsReady conditionsv1alpha1.ConditionType = "EndpointURLsReady"
+	// EndpointSliceReadyForURLs is a condition for APIExportEndpointSlice that reflects the readiness of the slice to
+	// provide URLs. It is set to True when the slice is ready to provide URLs which will be lifecycled based on existing consumers in the different
+	// shards.
+	APIExportEndpointSliceReadyForURLs conditionsv1alpha1.ConditionType = "EndpointReadyForURLs"
```

Comment:
`ReadyForEndpoints`.

Actually not clear why we need this. When is it not ready?

Reply:
Its more like handover/signal to secondary url controller. I used this a trigger. We could treat all conditions ready as signal too and just drop this all togheter.

Reply:
Was thinking about the label selector. It must be known, right?